config/v1alpha1/types_insights.go

```diff
@@ -48,17 +47,80 @@ type GatherConfig struct {
 	// When omitted, this means no opinion and the platform is left to choose a reasonable default, which is subject to change over time.
-	// The current default is None.
 	// +optional
 	DataPolicy DataPolicy `json:"dataPolicy,omitempty"`
 	// disabledGatherers is a list of gatherers to be excluded from the gathering. All the gatherers can be disabled by providing "all" value.
 	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The format for the disabledGatherer should be: {gatherer}/{function} where the function is optional.
+	// Gatherer consists of a lowercase string that may include underscores (_).
+	// Function consists of a lowercase string that may include underscores (_) and is separated from the gatherer by a forward slash (/).
 	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
 	// Run the following command to get the names of last active gatherers:
 	// "oc get insightsoperators.operator.openshift.io cluster -o json | jq '.status.gatherStatus.gatherers[].name'"
 	// An example of disabling gatherers looks like this: `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	// +kubebuilder:validation:MaxItems=100
+	// +optional
+	DisabledGatherers []DisabledGatherer `json:"disabledGatherers"`
+	// storageSpec is an optional field that allows user to define persistent storage for on-demand gathering
+	// jobs to store the Insights data archive.
+	// If omitted, the gathering job will use ephemeral storage.
+	// +optional
+	StorageSpec *StorageSpec `json:"storageSpec,omitempty"`
+}
+
+// disabledGatherer is a string that represents a gatherer that should be disabled
+// +kubebuilder:validation:MaxLength=256
+// +kubebuilder:validation:XValidation:rule=`self.matches("^[a-z]+[_a-z]*[a-z]([/a-z][_a-z]*)?[a-z]$")`,message=`disabledGatherer must be in the format of {gatherer}/{function} where the gatherer and function are lowercase strings that may include underscores (_) and are separated by a forward slash (/) if the function is provided`
+type DisabledGatherer string
+
+// storageSpec provides persistent storage configuration options for on-demand gathering jobs.
+// +kubebuilder:validation:XValidation:rule="has(self.type) && self.type == 'PersistentVolumeClaim' ?  has(self.persistentVolume) : !has(self.persistentVolume)",message="persistentVolume is required when type is PersistentVolumeClaim, and forbidden otherwise"
+type StorageSpec struct {
+	// type is a required field that specifies the type of storage that will be used to store the Insights data archive.
+	// Valid values are "PersistentVolumeClaim" and "Ephemeral".
+	// +required
+	Type StorageType `json:"type"`
+	// persistentVolume is an optional field that specifies the PersistentVolume that will be used to store the Insights data archive.
+	// The PersistentVolume must be created in the openshift-insights namespace.
```

Comment:
Reflect the discriminated union validation in the godoc for this field as well so that users of tooling like `oc explain ...` are informed of this validation.

Reply:
I don't think this has been properly resolved. Add a line in this godoc that states that this field is required when the type field is set to PersistentVolume and forbidden otherwise.

Users of tools like `oc explain` cannot see the CEL validation so it also needs to be explained in plain text in the godoc that is used to generate the field description.